sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageLockContainer.java

```diff
@@ -23,7 +23,7 @@
      * @return The updated value in the container. If the expiration time in the container is larger than
      *     {@code lockTokenExpiration}, then the current container value is used.
      */
-    public Instant addOrUpdate(UUID lockToken, Instant lockTokenExpiration) {
+    public Instant addOrUpdate(String lockToken, Instant lockTokenExpiration) {
```

Comment:
While  you are changing this to String, you can look at `receiveDeferredMessage` , there is a problem where we are not calling `addOrUpdate`  in `receiveDeferredMessage`  in PEEK LOCK mode. Thus once we receive DeferredMessage, it can not be Completed or deadLetter because MessageLockContainer does not have an entry for this message lock.